Runtime/Core/ApiGatewayManagement/AmazonGameLiftClientWrapper.cs

```diff
@@ -42,17 +58,181 @@ public async Task<SearchGameSessionsResponse> SearchGameSessions(SearchGameSessi
-            return new AmazonGameLiftClient(s_dummyAwsAccessKey, s_dummyAwsSecretKey, new AmazonGameLiftConfig
+            return await _amazonGameLiftClient.RegisterComputeAsync(request);
+        }
+
+        private async Task<GetComputeAuthTokenResponse> GetComputeAuthToken(GetComputeAuthTokenRequest request)
+        {
+            return await _amazonGameLiftClient.GetComputeAuthTokenAsync(request);
+        }
+
+        private Task<CreateFleetResponse> CreateFleet(CreateFleetRequest request)
+        {
+            return _amazonGameLiftClient.CreateFleetAsync(request);
+        }
+        
+        private RetriveAwsCredentialsResponse RetrieveAwsCredentials(string profileName)
+        {
+            var request = new RetriveAwsCredentialsRequest { ProfileName = profileName };
+            return _credentialsStore.RetriveAwsCredentials(request);
+        }
+        
+        public async Task UpdateAuthToken()
+        {
+            try
+            {
+                var config = _gameLiftClientSettings.GetGameLiftAnywhereConfiguration();
+                await GenerateAuthToken(config.ComputeName, config.FleetID);
+            }
+            catch (Exception e)
             {
-                ServiceURL = localEndpoint
-            });
+                Console.WriteLine(e);
+                throw;
+            }
         }
 
-        public async Task<DescribeGameSessionsResponse> DescribeGameSessions(DescribeGameSessionsRequest request)
+        private RetriveAwsCredentialsResponse SetupCredentials(string profileName)
         {
-            return await _amazonGameLiftClient.DescribeGameSessionsAsync(request);
+            var config = _gameLiftClientSettings.GetGameLiftAnywhereConfiguration();
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+            var credentials = RetrieveAwsCredentials(profileName);
+            return credentials;
+        }
+
+        private async Task GenerateAuthToken(string computeName, string fleetId)
+        {
+            try
+            {
+                var computeAuthTokenRequest = new GetComputeAuthTokenRequest
+                {
+                    ComputeName = computeName,
+                    FleetId = fleetId
+                };
+                var computeAuthTokenResponse = await GetComputeAuthToken(computeAuthTokenRequest);
+                _gameLiftClientSettings.authToken = computeAuthTokenResponse.AuthToken;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw;
+            }
         }
+
+        private async Task RegisterCompute(string computeName, string fleetId, string fleetLocation)
+        {
+            try
+            {
+                var ipAddress = Dns.GetHostEntry(Dns.GetHostName())
+                    .AddressList
+                    .FirstOrDefault(ip => ip.AddressFamily == AddressFamily.InterNetwork)?
+                    .ToString() ?? "0.0.0.0";
```

Comment:
Same as some of the above comments. IPAddress should be a parameter o this method, and the invoker can decide what it wants to pass in